cmake/modules/OmrUtility.cmake

```diff
@@ -139,43 +139,66 @@ function(omr_count_true out)
+		set(prop_bool "$<BOOL:${prop_genex}>")
+		set(out "$<${prop_bool}:${prop_genex}>$<$<NOT:${prop_bool}>:${out}>")
+	endforeach()
+	set(${output_var} "${out}" PARENT_SCOPE)
+endfunction()
+
+# omr_get_target_output_info(<target> <output_dir_var> <output_name_var>)
+#   Returns a pair of generator expressions.
+#   <output_dir_var> receives a generator expression which yields the output directory or <target>
+#   <output_name_var> receives a generator expression which yields the output name of <target>
+#   with the file suffix omitted.
+#   Note: currenly only executable and shared library targets are supported
+function(omr_get_target_output_info tgt output_dir_var output_name_var)
 	get_target_property(target_type ${tgt} TYPE)
 	if(target_type STREQUAL "EXECUTABLE")
-		get_target_property(target_directory ${tgt} RUNTIME_OUTPUT_DIRECTORY)
-		get_target_property(target_name      ${tgt} RUNTIME_OUTPUT_NAME)
-		if(NOT target_name)
-			set(target_name "${tgt}${CMAKE_EXECUTABLE_SUFFIX}")
-		endif()
+		set(search_prefixes "RUNTIME_" "")
 	elseif(target_type STREQUAL "SHARED_LIBRARY")
-		get_target_property(target_directory ${tgt} LIBRARY_OUTPUT_DIRECTORY)
-		get_target_property(target_name      ${tgt} LIBRARY_OUTPUT_NAME)
-		if(NOT target_name)
-			set(target_name "lib${tgt}${CMAKE_SHARED_LIBRARY_SUFFIX}")
-		endif()
+		set(search_prefixes "LIBRARY_" "")
 	else()
 		message(FATAL_ERROR "omr_get_target_path: ${tgt} type is ${target_type}")
 	endif()
-	set(${VAR} "${target_directory}/${target_name}" PARENT_SCOPE)
-endfunction(omr_get_target_path)
 
-# replace the suffix of ${input} with ${new_suffix}
-#
-# The suffix is defined here as the portion string
-# starting with the last "." in the last path segment.
-# This is consistent with:
-#   get_filename_component(suffix ${input} LAST_EXT)
-# without requiring cmake version 3.14.
-function(omr_replace_suffix VAR input new_suffix)
-	string(FIND "${input}" "/" slash_index REVERSE)
-	string(FIND "${input}" "." dot_index REVERSE)
-	if(dot_index GREATER slash_index)
-		string(SUBSTRING "${input}" 0 ${dot_index} base)
-		set(result "${base}${new_suffix}")
-	else()
-		set(result "${input}${new_suffix}")
-	endif()
-	set(${VAR} "${result}" PARENT_SCOPE)
-endfunction(omr_replace_suffix)
+	string(TOUPPER "${CMAKE_BUILD_TYPE}" build_type)
+	set(output_dirs)
+	set(output_names)
+	foreach(prefix IN LISTS search_prefixes)
+		if(build_type)
+			list(APPEND output_dirs  "${prefix}OUTPUT_DIRECTORY_${build_type}")
+			list(APPEND output_names "${prefix}OUTPUT_NAME_${build_type}")
+		endif()
+		list(APPEND output_dirs  "${prefix}OUTPUT_DIRECTORY")
+		list(APPEND output_names "${prefix}OUTPUT_NAME")
+	endforeach()
+
+	omr_genex_property_chain(dir_genex ${tgt} ${output_dirs} "$<TARGET_PROPERTY:${tgt},BINARY_DIR>")
+	omr_genex_property_chain(name_genex ${tgt} ${output_names} "${tgt}")
+	set(${output_dir_var} "${dir_genex}" PARENT_SCOPE)
+	set(${output_name_var} "$<TARGET_PROPERTY:${tgt},PREFIX>${name_genex}" PARENT_SCOPE)
```

Comment:
This doesn't include the prefix in `output_name_var`: e.g. we get `omrsig` instead of `libomrsig`.

It also omits a desired part of the suffix: in OpenJ9 the shared library suffix is `29.so` (the `29` part is missing) so we get `j9vm` instead of `libj9vm29`. Perhaps the right way to fix this problem is to update OpenJ9 to avoid modifying `CMAKE_SHARED_LIBRARY_SUFFIX` by specifying `OUTPUT_NAME` in calls to `omr_add_library`.

Reply:
see eclipse/openj9#9619 (which is currently waiting for the `OUTPUT_NAME` change to promote)

Reply:
prefix issue should now also be resolved

Reply:
I think this should wait until eclipse/openj9#9619 has landed.

Reply:
I don't think it can. I need to double check but this would fix the build failure which is blocking the omr-acceptance builds which we need to get #5218 so that  eclipse/openj9#9619  can be merged

Reply:
On the other hand, merging this before eclipse/openj9#9619 will break (cmake) builds by not producing debuginfo files with the right names.

Maybe this needs to temporarily include the portion of the suffix before the '.'.

Reply:
Actually it looks like we don't do cmake osx for the acceptance build, only the nightlies, and it looks like #5128 has promoted

Reply:
So let's plan to get eclipse/openj9#9619 merged before merging this so as not to break builds.